bark/models/observer/observer_model_parametric.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2020 fortiss GmbH
+//
+// Authors: Julian Bernhard, Klemens Esterle, Patrick Hart and
+// Tobias Kessler
+//
+// This work is licensed under the terms of the MIT license.
+// For a copy, see <https://opensource.org/licenses/MIT>.
+
+
+#include <memory>
+#include "bark/models/observer/observer_model_parametric.hpp"
+#include "bark/commons/distribution/distribution.hpp"
+#include "bark/commons/transformation/frenet_state.hpp"
+#include "bark/world/observed_world.hpp"
+
+namespace bark {
+namespace models {
+namespace observer {
+using namespace bark::world;
+using namespace bark::world::objects;
+using namespace bark::commons::transformation;
+using namespace bark::commons;
+
+ObserverModelParametric::ObserverModelParametric(bark::commons::ParamsPtr params) :
+  ObserverModel(params),
+  ego_state_deviation_dist_(params->GetDistribution("ObserverModelParametric::EgoStateDeviationDist",
+                                  "From what distribution is the ego frenet state deviations "
+                                  "sampled, must have dimension = 5",
+                                  "MultivariateDistribution")),
+  others_state_deviation_dist_(params->GetDistribution("ObserverModelParametric::OtherStateDeviationDist",
+                                  "From what distribution is the others frenet state deviation"
+                                  "sampled, must have dimension = 5",
+                                  "MultivariateDistribution")) {}
+
+ObserverModelParametric::ObserverModelParametric(const ObserverModelParametric& observer_model) :
+      ObserverModel(observer_model.GetParams()),
+      ego_state_deviation_dist_(observer_model.ego_state_deviation_dist_),
+      others_state_deviation_dist_(observer_model.others_state_deviation_dist_) {}
+
+
+ObservedWorld ObserverModelParametric::Observe(
+    const WorldPtr& world, const AgentId& agent_id) {
+  
+  // Clone world here since otherwise we change global world state
+  auto observed_world = ObservedWorld(world->Clone(), agent_id);
+
+  AddStateDeviationFrenet(observed_world.GetEgoAgent(), ego_state_deviation_dist_);
+  for (auto& agent : observed_world.GetOtherAgents()) {
```

Comment:
do we only want the possibility of the other agents to be perturbed or should we also introduce it for the ego-agent, e.g. modeling a poor localization?

Reply:
See line 47, where this is implemented